src/client.ts

```diff
@@ -4535,11 +4535,18 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
             return Promise.resolve({}); // guests cannot send receipts so don't bother.
         }
 
-        const path = utils.encodeUri("/rooms/$roomId/receipt/$receiptType/$eventId", {
+        let path = utils.encodeUri("/rooms/$roomId/receipt/$receiptType/$eventId", {
             $roomId: event.getRoomId(),
             $receiptType: receiptType,
             $eventId: event.getId(),
         });
+
+        if (event.threadRootId) {
+            path += utils.encodeUri("/$threadId", {
+                $threadId: event.threadRootId,
+            });
+        }
```

Comment:
We should figure out if we want to keep sending this as part of the path (or the body or a GET parameter), but this matches the Synapse proof-of-concept for now. See https://github.com/matrix-org/matrix-spec-proposals/pull/3771#discussion_r918213697.

Reply:
Changed to match what the MSC says